contrib/dokku-installer.rb

```diff
@@ -3,21 +3,26 @@
 require "open3"
 
 if ARGV[0] == "onboot"
-  File.open("/etc/init/dokku-installer.conf", "w") do |f|
-    f.puts "start on runlevel [2345]"
-    f.puts "exec #{File.absolute_path(__FILE__)} selfdestruct"
-  end if File.directory?("/etc/init")
-  File.open("/etc/systemd/system/dokku-installer.service", "w") do |f|
-    f.puts "[Unit]"
-    f.puts "Description=Dokku web-installer"
-    f.puts ""
-    f.puts "[Service]"
-    f.puts "ExecStart=#{File.absolute_path(__FILE__)} selfdestruct"
-    f.puts ""
-    f.puts "[Install]"
-    f.puts "WantedBy=multi-user.target"
-    f.puts "WantedBy=graphical.target"
-  end if File.directory?("/etc/systemd/system")
+  if File.realpath("/sbin/init") != "/sbin/init" then
```

Comment:
Yes, this code doesnt work on systems with SysV-init, because there is no systemd directory, where to place dokku service. In case SysV-init initscript is installed in other cases systemd is used.